baybe/campaign.py

```diff
@@ -282,6 +285,10 @@ def recommend(
                 f"{batch_size=}."
             )
 
+        # Invalidate cached recommendation if pending measurements are provided
```

Comment:
Assuming 2 subsequent calls. You'd get the same recommendations as for your first call, even if your second call had pending points, simply because the cached ones would be returned. Assuming the pending points were created based on the first batch of recommendations its likely that the recommendations from a second call even overlap with the already existing pending points - a calamity.

Imo pending points here take precedence and it should be ensured they recommendations correctly incorporate that. 

One could also think about cached pending points which would play into the whole cached retrieval, but I was quite clear in the PR description that this PR simply delivers the pending points solution in a sateless manner, which can be extended if really desired later on

Reply:
@AVHopp requiring your response or resolution